tests/test_table.py

```diff
@@ -1055,6 +1056,43 @@ def test_indexed_table_mixin():
     assert table.fast_slice(2, 13) == pa_table.slice(2, 13)
 
 
+@pytest.mark.parametrize(
+    "arrays",
+    [
+        [pa.array([[1, 2, 3, 4]]), pa.array([[10, 2]])],
+        [
+            pa.array([[[1, 2], [3]]], pa.list_(pa.list_(pa.int32()), 2)),
+            pa.array([[[10, 2, 3], [2]]], pa.list_(pa.list_(pa.int32()), 2)),
+        ],
```

Comment:
Can you add some tests with sliced arrays ? They may have offsets that don't start at 0 IIRC.
And maybe arrays with None as well.

Reply:
Done! 

> And maybe arrays with None as well.

We already have a test for this case